packages/design-system/src/components/Autocomplete/Autocomplete.tsx

```diff
@@ -232,7 +233,8 @@ export const Autocomplete = (props: AutocompleteProps) => {
       : undefined,
     onSelectionChange: onChange
       ? (selectedKey) => {
-          const selectedItem = items ? items.find((item) => selectedKey === item.id) : undefined;
+          const selectedItem =
+            items && selectedKey != null && findItemById(String(selectedKey), items);
```

Comment:
```suggestion
            items ? findItemById(String(selectedKey), items) : undefined;
```
Do we need the `selectedKey != null` check here? Seems like we're only reaching this point if `selectedKey` is defined (L235)